index.js

```diff
@@ -4,32 +4,34 @@ const fs = require('fs');
 
 /**
  * Converts a callback style call to a Promise
- * @param  {function} f
- * @param  {object[]} args    arguments to pass to the function when invoking it
- * @return {Promise<object>}  object returned by the function
+ * @param  {function} f       a node style function that accepts a callback
+ * @param  {Object[]} args    arguments to pass to the function when invoking it
+ * @return {Promise<Object>}  object returned by the function
  */
 function promisify(f, args) {
   return new Promise((resolve, reject) => f.apply(this, args.concat((err, x) => err ? reject(err) : resolve(x))));
 }
 
 /**
- * Get a kernel resources object 
- * @param  {object}   kernelInfo              description of a kernel
+ * Get a kernel resources object
+ * @param  {Object}   kernelInfo              description of a kernel
  * @param  {string}   kernelInfo.name         name of the kernel
  * @param  {string}   kernelInfo.resourceDir  kernel's resources directory
- * @return {Promise<object>}                  Promise for a kernelResources object
+ * @return {Promise<Object>}                  Promise for a kernelResources object
  */
 function getKernelResources(kernelInfo) {
-  return Promise.resolve(kernelInfo).then(kernelInfo => 
+  return Promise.resolve().then(() =>
```

Comment:
@rgbkrk This is wrong, it probably cause things to break.  `Promise.resolve` can take a Promise<Object> or an Object and it will create a Promise that resolves to the object.  IOW, by using this method, I was making it so the function could handle both Promises and synchronous input.